iothub/service/src/JobProperties.cs

```diff
@@ -52,49 +53,121 @@ public JobProperties()
 
         /// <summary>
-        /// The name of the blob that will be created in the provided output blob container.  This blob will contain
+        /// The name of the blob that will be created in the provided output blob container. This blob will contain
         /// the exported device registry information for the IoT Hub.
         /// </summary>
+        /// <remarks>
+        /// If not specified, defaults to "devices.txt"
+        /// </remarks>
         [JsonProperty(PropertyName = "outputBlobName", NullValueHandling = NullValueHandling.Ignore)]
         public string OutputBlobName { get; set; }
 
         /// <summary>
-        /// Optional for export jobs; ignored for other jobs.  Default: false.  If false, authorization keys are included
-        /// in export output.  Keys are exported as null otherwise.
+        /// Optional for export jobs; ignored for other jobs. Default: false. If false, authorization keys are included
+        /// in export output. Keys are exported as null otherwise.
         /// </summary>
         [JsonProperty(PropertyName = "excludeKeysInExport", NullValueHandling = NullValueHandling.Ignore)]
         public bool ExcludeKeysInExport { get; set; }
 
         /// <summary>
-        /// System genereated.  Ignored at creation.
-        /// If status == failure, this represents a string containing the reason.
+        /// Specifies authentication type being used for connecting to storage account
         /// </summary>
-        [JsonProperty(PropertyName = "failureReason", NullValueHandling = NullValueHandling.Ignore)]
-        public string FailureReason { get; set; }
+        [JsonProperty(PropertyName = "storageAuthenticationType", NullValueHandling = NullValueHandling.Ignore)]
+        public StorageAuthenticationType StorageAuthenticationType { get; set; }
+
+#pragma warning disable CA1054 // Uri parameters should not be strings
+
+        /// <summary>
+        /// Creates an instance of JobProperties with paramters ready to start an Import job
+        /// </summary>
+        /// <param name="inputBlobContainerUri">URI containing SAS token to a blob container that contains registry data to sync</param>
+        /// <param name="outputBlobContainerUri">URI containing SAS token to a blob container.  This is used to output the status of the job and the results.</param>
+        /// <param name="inputBlobName">The blob name to be used when importing from the provided input blob container</param>
+        /// <param name="storageAuthenticationType">Specifies authentication type being used for connecting to storage account</param>
+        /// <returns>An instance of JobProperties</returns>
+        public static JobProperties CreateForImportJob(
+            string inputBlobContainerUri,
+            string outputBlobContainerUri,
+            string inputBlobName = null,
+            StorageAuthenticationType storageAuthenticationType = StorageAuthenticationType.KeyBased)
```

Comment:
Factory method to help users know which parameters are required and which are optional for import job. Method below for export job.